pkgs/top-level/all-packages.nix

```
@@ -16245,15 +16245,57 @@ with pkgs;
   gcc-arm-embedded-12 = callPackage ../development/compilers/gcc-arm-embedded/12 { };
   gcc-arm-embedded = gcc-arm-embedded-12;
 
-  # It would be better to match the default gcc so that there are no linking errors
-  # when using C/C++ libraries in D packages, but right now versions >= 12 are broken.
-  gdc = gdc11;
+  # Has to match the default gcc so that there are no linking errors when
+  # using C/C++ libraries in D packages
+  gdc = gdc13;
   gdc11 = wrapCC (gcc11.cc.override {
     name = "gdc";
-    langCC = false;
-    langC = false;
+    langCC = true; # required to bootstrap newer gdc
```

```suggestion
    langCC = true; # required to bootstrap newer gdc
```

It would be better to make this an `assert` instead of a comment.